File: WIP/1409-Queries_on_a_Permutation_With_key.ts
function processQueries(queries: number[], m: number): number[] {
  // Contains values from 1 to m.
  let p: number[] = [];

  // Holds position of query[i] in p
  let position: number[] = [];

  // Populates p with values from i to m
  for (let i = 1; i <= m; i++) {
    p.push(i);
  }

  console.log(`Initial p is ${p}`);

  for (let i = 0; i < queries.length; i++) {
    position.push(p.indexOf(queries[i]));

    let temp = p.splice(queries[i]);
    p.splice(0, 0, temp[0]);

    console.log(
      `The value of p is \t${p} \tafter moving \t${i} \tto index 0 and return value is \t${position}`
    );
  }

  console.log(position);
  return position;
}

processQueries([3, 1, 2, 1], 5);

File: random_solutions/Medium/1409-Queries_on_a_Permutation_With_Key.py
class Solution:
    # processQueries    -> Time { O(n), θ(n), Ω(1) } and Space { O(n), θ(n), Ω(n) }

    def processQueries(self, queries: list[int], m: int) -> list[int]:
        p = list(range(1, m + 1))
        position = []

        for query in queries:
            ndx = p.index(query)
            position.append(ndx)

            value = p.pop(ndx)
            p.insert(0, value)

        return position
